src/dom/Decimal.ts

```diff
@@ -27,13 +27,44 @@ export class Decimal extends Value(
 
   /**
    * Constructor.
-   * @param value         The numeric value to represent as a decimal.
+   * @param value         The Ion decimal value to represent as a decimal.
    * @param annotations   An optional array of strings to associate with `value`.
    */
-  constructor(value: ion.Decimal, annotations: string[] = []) {
-    super(value.numberValue());
-    this._decimalValue = value;
-    this._numberValue = value.numberValue();
+  constructor(value: ion.Decimal, annotations?: string[]);
+
+  /**
+   * Constructor.
+   * @param value         The string value to represent as a decimal.
+   * @param annotations   An optional array of strings to associate with `value`.
+   */
+  constructor(value: string, annotations?: string[]);
+
+  /**
+   * Constructor.
+   * @param value         The number value to represent as a decimal.
+   * @param annotations   An optional array of strings to associate with `value`.
+   */
+  constructor(value: number, annotations?: string[]);
+
+  // This is the unified implementation of the above signatures and is not visible to users.
+  constructor(
+    value: ion.Decimal | string | number,
+    annotations: string[] = []
+  ) {
+    if (typeof value === "string") {
+      super(Number(value));
+      this._decimalValue = new ion.Decimal(value);
+      this._numberValue = Number(value);
```

Comment:
We can avoid doing the same conversion twice:
```suggestion
      let numberValue = Number(value);
      super(numberValue);
      this._decimalValue = new ion.Decimal(value);
      this._numberValue = numberValue;
```
I'm unsure if there's a benefit/drawback to doing `Number(value)` (`string` -> `Number`) over `this._decimalValue.numberValue()` (`string` -> `decimal` -> `Number`). The latter is more steps, but may preserve precision better? I'm not terribly concerned about it because it only affects the `Number` approximation of the `Decimal`, which will definitely preserve precision.